lib/pf/Switch/AeroHIVE/AP_http.pm

```diff
@@ -62,27 +62,32 @@ Overriding the default implementation for the external captive portal
 =cut
 
 sub returnRadiusAccessAccept {
-    my ($self, $vlan, $mac, $port, $connection_type, $user_name, $ssid, $wasInline, $user_role) = @_;
+    my ($self, $args) = @_;
     my $logger = $self->logger;
 
     my $radius_reply_ref = {};
 
-    my $node = node_view($mac);
+    my $node = $args->{'node_info'};
+
+    my $filter = pf::access_filter::radius->new;
+    my $rule = $filter->test('returnRadiusAccessAccept', $args);
 
-    my $violation = pf::violation::violation_view_top($mac);
+    my $violation = pf::violation::violation_view_top($args->{'mac'});
     # if user is unregistered or is in violation then we reject him to show him the captive portal 
     if ( $node->{status} eq $pf::node::STATUS_UNREGISTERED || defined($violation) ){
-        $logger->info("[$mac] is unregistered. Refusing access to force the eCWP");
+        $logger->info("[$args->{'mac'}] is unregistered. Refusing access to force the eCWP");
         my $radius_reply_ref = {
             'Tunnel-Medium-Type' => $RADIUS::ETHERNET,
             'Tunnel-Type' => $RADIUS::VLAN,
             'Tunnel-Private-Group-ID' => -1,
         }; 
+        $radius_reply_ref = $filter->handleAnswerInRule($rule,$args,$radius_reply_ref);
         return [$RADIUS::RLM_MODULE_OK, %$radius_reply_ref]; 
 
     }
     else{
-        $logger->info("[$mac] Returning ACCEPT");
+        $logger->info("[$args->{'mac'}] Returning ACCEPT");
+        $radius_reply_ref = $filter->handleAnswerInRule($rule,$args,$radius_reply_ref);
```

Comment:
I think you should wrap this in a subroutine and handle the 'unfilter' flag in it so you don't have to do it in every returnRadiusAccessAccept